common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -703,6 +715,14 @@ private synchronized KeyPair generateKeyPairFromAndroidKeyStore()
         }
     }
 
+    public static synchronized void applyKeyStoreLocaleWorkarounds(@NonNull final Locale currentLocale) {
```

Comment:
The synchronization here needs to encompass the entire time that the locale is altered.  Otherwise a second thread entering this section will pick the current locale as the new altered locale, the first thread retires and restores the new locale and the second sets it to it's view of the default locale, which is the wrong one.